espnet2/train/spk_trainer.py

```diff
@@ -72,41 +72,65 @@ def validate_one_epoch(
 
         scores = []
         labels = []
+        spk_embd_dic = {}
+        bs = 0
+        start_time = time.time()
+
         # [For distributed] Because iteration counts are not always equals between
         # processes, send stop-flag to the other processes if iterator is finished
         iterator_stop = torch.tensor(0).to("cuda" if ngpu > 0 else "cpu")
+
+        # fill dictionary with speech samples
+        utt_id_list = []
+        speech_list = []
         for utt_id, batch in iterator:
+            bs = max(bs, len(utt_id))
             assert isinstance(batch, dict), type(batch)
+            for _utt_id, _speech, _speech2 in zip(
+                utt_id, batch["speech"], batch["speech2"]
+            ):
+                _utt_id_1, _utt_id_2 = _utt_id.split("*")
+                if _utt_id_1 not in utt_id_list:
+                    utt_id_list.append(_utt_id_1)
+                    speech_list.append(_speech)
+                if _utt_id_2 not in utt_id_list:
+                    utt_id_list.append(_utt_id_2)
+                    speech_list.append(_speech2)
+
```

Comment:
How about adding logging to report some statistics?
- at least some counts (e.g., speaker embedding vectors, how many similarities)
- other statistics, e.g., the global mean/variance of the speaker embedding vectors, scores, duration, etc.

This is just a suggestion, but this kind of statistics makes us easy to detect some irregular behaviors in advance.

Reply:
Also, if we provide logging before/after speaker embedding/scoring computations give us more information about the computational cost.